### ISSUE TITLE
What is the reason for moving from doing setIn or updateIn to recursive solution?

### ISSUE THREAD

**Issue by abhinavsinghvi** [intelie/immutable-js-patch#8]
Hi, 

   I have a question related to this library, what is the reason for moving from setIn to recursively traversing through the path and setting the value? Doesn't it make it complicated?

Thanks,
Abhinav

**Comment by pedroteixeira** [comment]
I agree, recursion is not needed and the code could probably leverage existing setIn or updateIn, if one can still have acess to the last key when a path does not exist - because it's used to infer the best type (Immutable.List() or Immutable.Map()) for a new object.

Note that this implementation does not use a json-schema to aid on creating new objects when applying patches on undefined values, which could lead to issues. e.g: "/a/b/0/" is b a List or Map?
